cpu/pic32/lib/pic32_pwm.c

```diff
@@ -0,0 +1,187 @@
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ * contributors may be used to endorse or promote products derived from this
+ * software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#ifdef __USE_PWM__
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <p32xxxx.h>
+#include <pic32_clock.h>
+#include <pic32_pwm.h>
+/*----------------------------------------------------------------------------------------------*/
+static const uint32_t prescaler_values[] = {1, 2, 4, 8, 16, 32, 64, 256};
+/*----------------------------------------------------------------------------------------------*/
+#define PWM(XX, YY)                                                                                                 \
+    /**                                                                                                             \
+     * @brief Configure timer parameters                                                                            \
+     *                                                                                                              \
+     * PR and prescalers of timer need to be found such that the period of the timer is as close as possible        \
+     * from the desired period. At the same time, PR needs to be as high as possible such that the resolution       \
+     * is as high as possible.                                                                                      \
+     *                                                                                                              \
+     * @return 0 if successful                                                                                      \
+     */                                                                                                             \
+    static uint8_t configure_timer(uint32_t period)                                                                 \
```

Comment:
Move outside of the define, otherwise it won't compile with multiple defines. This seems to be (apart from the assigns) a helper function.